### DISCUSSION TITLE
Error on the ESlint when trying of npm run build

### DISCUSSION THREAD

**Discussion by gorokhovPavel** [react-hook-form/react-hook-form#4625] [Q&A]
Hello! I have a problem with your library when i was trying to run command npm run build 

There were errors in the file /node_modules/react-hook-form/dist/types/utils.d.ts in the esLInt

For example, 
> line (24:77) the : declare type PathImpl<K extends string | number, V> = V extends Primitive ? `${K}` : `${K}` | `${K}.${Path<V>}`;
> error : ESLint: Parsing error: '?' expected.

And also others errors like that on esLint

versions of packages are 

```json
"eslint": "7.22.0",
"react-hook-form": "7.0.0",
```

Would you please share any advices?

**Comment by jorisre** [comment]
Hi @gorokhovPavel 

Can you provide a codesandbox or a repo with your problem please?

**Comment by 4ortytwo** [comment]
I'm using next.js and deploying to vercel and had this error just now, read around and found out that if you use RHF v7, you should have  TS 4.1 above, I upgraded typescript to latest and the error is gone. Hope it helps someone.

**Reply by bluebill1049** [reply]
thanks for sharing solution.
